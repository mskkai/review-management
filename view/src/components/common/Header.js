import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../../actions/auth";
import { Button } from "primereact/button";
import { VscFeedback } from "react-icons/vsc";
import { FaUserAlt } from "react-icons/fa";

export const Header = ({ user, startLogout }) => {
  return (
    <div className="p-shadow-6 header">
      <div className="header-items">
        <div className="header-items__user-details" title={user.name}>
          <span className="header-items__user-name">
            {user.avatar ? (
              <img src={user.avatar} alt=""></img>
            ) : (
              <FaUserAlt className="header-items__user-image" />
            )}

            {user.name}
          </span>
        </div>

        <div className="header-items__title">
          <Link to="/home">
            <div className="header-items__title-icon">
              <VscFeedback />
            </div>
          </Link>
          REVIEW MANAGEMENT
        </div>

        <Button
          label="SIGN OUT"
          icon="pi pi-sign-out header-items__title-button-icon"
          className="header-items__title-button p-button-secondary p-button-text show-for-desktop"
          iconPos="right"
          onClick={startLogout}
        />
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};
const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
