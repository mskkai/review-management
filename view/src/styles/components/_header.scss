.header {
  background: $color-light-varient-2;
  height: 5rem;
  color: $blue-mid;
  border-bottom: 1px solid $color-light-varient-1;
}

.header-items {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-left: 2rem;
  padding-right: 2rem;
}

.header-items__title {
  font-family: $font-app;
  font-size: $font-size-large;
  text-decoration: none;
  padding-top: 1rem;
  letter-spacing: 0.1rem;
}

.header-items__title-icon {
  display: inline-block;
  vertical-align: middle;
  margin-right: 0.8rem;
  font-size: $font-size-xlarge;
  color: $yellow-mid;
  bottom: 0.2rem;
  position: relative;
}

.header-items__title-button {
  font-size: $font-size-medium !important;
  margin-bottom: 1rem;
  font-weight: 500 !important;
}

.header-items__title-button-icon {
  font-weight: 900 !important;
}

.header-items__user-details {
  width: 12%;
}

.header-items__user-image {
  width: 2rem;
  height: 2rem;
  border-radius: 2rem;
  position: relative;
  top: 0.3rem;
  right: 1rem;
}

.header-items__user-name {
  display: block;
  font-size: $font-size-large;
  letter-spacing: 0.2rem;
  text-transform: capitalize;
  text-align: center;
  color: $blue-lightish;
  font-weight: 600;
  height: 4rem;
  padding: 0.8rem 2rem 0.5rem 2rem;
  border-radius: 4px;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  img {
    @extend .header-items__user-image;
  }
}
