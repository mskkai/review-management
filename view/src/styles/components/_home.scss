.home-container {
  display: flex;
  align-items: center;
  justify-content: center;
  height: calc(100vh - 12rem);
}

.home-container__title {
  font-style: italic;
  font-size: 5rem;
  text-transform: uppercase;
}

.home__carousel {
  .home__carousal-item-header {
    color: $blue-dark;
    font-family: "Dancing Script" !important;
  }

  .home__carousal-item {
    .home__carousal-item-content {
      border: 1px solid var(--surface-d);
      border-radius: 3px;
      margin: 0.3rem;
      text-align: center;
      padding: 2rem 0;
    }

    .home__carousal-item-image {
      font-size: 10rem;
    }

    .home__carousal-item-button-text {
      font-size: $font-size-large !important;
    }
    .home__carousal-item-text {
      font-size: $font-size-medium !important;
    }
    .home__carousal-item-link {
      text-decoration: none;
    }
  }
}
