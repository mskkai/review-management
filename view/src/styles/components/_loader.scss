.loader {
  display: flex;
  align-items: center;
  justify-items: center;
  width: calc(100vw - 6rem);
  height: calc(100vh - 12rem);

  .p-progress-spinner-circle {
    stroke-width: 4;
  }

  .p-progress-spinner {
    width: 70px;
  }
}
