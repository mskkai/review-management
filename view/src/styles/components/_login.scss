.box-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
  background-image: url("/images/background_1.jpg");
  background-size: cover;
  width: 100vw;
  height: 100vh;
}

.box-container__item {
  display: flex;
  border-radius: 0.5rem;
  background: $color-lightish;
  min-width: 30vw;
  height: 50vh;
  text-align: center;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  margin: 2em;
  padding: 2em;
}

.box-container__text {
  text-align: center;
}

.box-container__header {
  @extend .box-container__text;
  font-size: $font-size-xlarge;
  line-height: 1;
  margin: 0 1rem;
  text-transform: uppercase;
}

.box-container__sub-header {
  @extend .box-container__text;
  font-size: $font-size-medium;
  margin: 0 1rem;
  text-align: center;
  text-transform: capitalize;
}

.box-container__button-icon {
  font-size: $font-size-large;
  margin-left: 1rem;
  text-transform: uppercase;
}
