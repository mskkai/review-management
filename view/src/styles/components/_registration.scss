.registration-view {
  width: 50vw;
  margin: 0 auto;
}

.registration__tab-view {
  @extend .registration-view;
  padding: 2rem;
  font-size: $font-size-lighter !important;

  i,
  span {
    vertical-align: middle;
    font-size: $font-size-lighter !important;
  }

  span {
    margin: 0 0.5rem;
  }

  .p-button {
    margin-right: 0.25rem;
  }

  .p-tabview p {
    line-height: 1.5;
    margin: 0;
  }
}

.registration__progress-bar {
  @extend .registration-view;
}

.registration__button {
  height: 5rem;
  align-self: center;
  font-weight: bold !important;
  width: max-content;
}

.registration__button-right {
  @extend .registration__button;
  justify-self: end;
}

.registration__button-left {
  @extend .registration__button;
  justify-self: start;
}
