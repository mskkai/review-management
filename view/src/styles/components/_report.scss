.report__nav-header {
  margin: 0.5rem;
}

.report__button-link {
  text-decoration: none;
}

.report-header {
  display: flex;
  justify-content: space-between;
}

.report-header__multi-select {
  width: 20rem;
  font-size: $font-size-small !important;
  text-transform: uppercase;
  padding: 0.1rem;
}

.report-header__button {
  font-size: $font-size-lighter !important;
}

.report-body__dialog-icon {
  font-size: 2rem;
}
