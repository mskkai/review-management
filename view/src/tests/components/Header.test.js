import React from "react";
import { Header } from "../../components/Header";
import { shallow } from "enzyme";

test("should render Header correctly using react test renderer", () => {
  const wrapper = shallow(<Header startLogout={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test("should logout on button click", () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find("button").simulate("click");
  expect(startLogout).toHaveBeenCalled();
});
