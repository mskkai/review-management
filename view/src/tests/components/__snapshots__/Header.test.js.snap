// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly using react test renderer 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        to="/dashboard"
      >
        <h1>
          React App
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
